src/Protocol/ProtocolRecognizer.cpp

```diff
@@ -76,6 +76,39 @@ AString cMultiVersionProtocol::GetVersionTextFromInt(cProtocol::Version a_Protoc
 
 
 
+bool cMultiVersionProtocol::CheckHTTPRequest(cClientHandle & a_Client, ContiguousByteBuffer & a_Data)
```

Comment:
The name here is off, this should be TryHandleHTTPRequest (better yet, separate out the checking and the handling into two functions.